app/validations/address.js

```diff
@@ -36,5 +36,13 @@ export function getAddressValidations(isAlwaysRequired = false) {
         ? null
         : ['street', 'number', 'postcode', 'province'],
     }),
+    province: validatePresence({
+      presence: isAlwaysRequired,
```

Comment:
This won't work as expected. The wording of this property is confusing. If set to false, this will only be valid if the field doesn't contain any data. But because of the `on` config, that will only trigger if any of the listed fields have data.

So if you enter a street name, and also select a province, the province will show an error.

For now it seems that the only places that require a province also have `isAlwaysRequired` set to true so we can take a shortcut and use that flag for the province as well. (That will fall apart if we ever need an `optional` address that requires a province if some address data is filled, but for now that's not the case).

Adding this province object conditionally based on the `isAlwaysRequired` should do the trick I think.
Pseudo / untested:
```js
let addressValidations = { //all validations but not the province one }

let isProvinceRequired = isAlwaysRequired;
if (isProvinceRequired) {
  addressValidations.province = validatePresence({
    presence: true,
    ignoreBlank: true,
    message: REQUIRED_MESSAGE,
  });
}

return addressValidations;
```

Reply:
thx, i'll test it and let you know